tests/integration/integration_python_acceptance_test.go

```diff
@@ -85,6 +85,17 @@ func TestDynamicPython(t *testing.T) {
 			Additive: true,
 			ExtraRuntimeValidation: func(t *testing.T, stack integration.RuntimeValidationStackInfo) {
 				assert.Equal(t, randomVal, stack.Outputs["random_val"].(string))
+
+				// Regression testing the workaround for https://github.com/pulumi/pulumi/issues/8265
+				// Ensure the __provider input was marked secret
+				dynRes := stack.Deployment.Resources[2]
+				providerVal := dynRes.Inputs["__provider"]
```

Comment:
Should we check the Output property to ensure it is also a secret?

Reply:
Can do